dashboard/app/controllers/courses_controller.rb

```diff
@@ -43,7 +43,25 @@ def create
     end
   end
 
+  def update
+    course = Course.find_by_name!(params[:course_name])
+    course.update_and_persist(params[:scripts], i18n_params)
```

Comment:
A pattern we use elsewhere is to have an `after_save :write_serialization` vs. calling the serialization explicitly.